Source/Flow/Private/FlowAsset.cpp

```diff
@@ -189,35 +189,73 @@ void UFlowAsset::HarvestNodeConnections()
 	}
 }
 
-void UFlowAsset::AddCustomInput(const FName& InName)
+void UFlowAsset::AddCustomInput(const FName& EventName)
 {
-	if (!CustomInputs.Contains(InName))
+	if (!CustomInputs.Contains(EventName))
 	{
-		CustomInputs.Add(InName);
+		CustomInputs.Add(EventName);
 	}
 }
 
-void UFlowAsset::RemoveCustomInput(const FName& InName)
+void UFlowAsset::RemoveCustomInput(const FName& EventName)
 {
-	if (CustomInputs.Contains(InName))
+	if (CustomInputs.Contains(EventName))
 	{
-		CustomInputs.Remove(InName);
+		CustomInputs.Remove(EventName);
 	}
 }
+
+void UFlowAsset::AddCustomOutput(const FName& EventName)
+{
+	check(!CustomOutputs.Contains(EventName));
+
+	CustomOutputs.Add(EventName);
+}
+
+void UFlowAsset::RemoveCustomOutput(const FName& EventName)
+{
+	check(CustomOutputs.Contains(EventName));
+
+	CustomOutputs.Remove(EventName);
+}
 #endif // WITH_EDITOR
 
+UFlowNode_CustomInput* UFlowAsset::TryFindCustomInputNodeByEventName(const FName& EventName) const
+{
+	for (UFlowNode_CustomInput* InputNode : CustomInputNodes)
+	{
+		if (IsValid(InputNode) && InputNode->GetEventName() == EventName)
+		{
+			return InputNode;
+		}
+	}
+
+	return nullptr;
+}
+
 UFlowNode* UFlowAsset::GetDefaultEntryNode() const
 {
+	UFlowNode* FirstStartNode = nullptr;
+
 	for (const TPair<FGuid, UFlowNode*>& Node : Nodes)
 	{
 		UFlowNode_Start* StartNode = Cast<UFlowNode_Start>(Node.Value);
-		if (StartNode && StartNode->GetConnectedNodes().Num() > 0)
+		if (StartNode)
 		{
-			return StartNode;
+			if (StartNode->GetConnectedNodes().Num() > 0)
+			{
+				return StartNode;
+			}
+			else if (!FirstStartNode)
+			{
+				FirstStartNode = StartNode;
+			}
 		}
 	}
 
-	return nullptr;
+	// If none of the found start nodes have connections, 
+	//  fallback to the first start node we found
+	return FirstStartNode;
```

Comment:
If none of the start nodes had any connected nodes (which is often the case for our graphs) then falling back to the first start node discovered.

We only use "Start" node for one-time initialization.  Most of the connected nodes are connected to custom inputs, which aren't Start nodes. 

After integrating with latest, but without this fix, our flow graphs never started (because nothing was connected to the start node).

Reply:
Very useful fix, thanks :)